src/kubeobjects/secret.go

```diff
@@ -47,15 +47,22 @@ func (query SecretQuery) Update(secret corev1.Secret) error {
 }
 
 func (query SecretQuery) CreateOrUpdate(secret corev1.Secret) error {
-	err := query.Create(secret)
-
-	if !k8serrors.IsAlreadyExists(err) {
+	currentSecret, err := query.Get(types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			query.log.Info("secret not found, creating", "name", secret.Name, "namespace", secret.Namespace)
+			return errors.WithStack(query.Create(secret))
+		}
 		return errors.WithStack(err)
 	}
 
-	query.log.Info("secret already exists", "name", secret.Name, "namespace", secret.Namespace)
+	if !AreSecretsEqual(secret, currentSecret) {
+		return errors.WithStack(query.Update(secret))
```

Comment:
If the above logs the creation of the secret, the update should probably also be logged